### PR TITLE
October 2020 updates

### PR DESCRIPTION
**Pull request by shirolkar**
Updates for October 2020. Issues addressed:
#28 #30 #33 #34 #35

### CODE REVIEW

File: Hands-on lab/HOL step-by-step - MLOps.md [hunk 1]
```diff
@@ -79,13 +94,13 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
+In this hands-on lab, you will learn how Wide World Importers (WWI) can leverage Deep Learning technologies to scan through their vehicle specification documents to find compliance issues with new regulations. You will standardize the model format to ONNX and observe how this simplifies inference runtime code, enabling pluggability of different models and targeting a broad range of runtime environments, and most importantly, improves inferencing speed over the native model. You will build a DevOps pipeline to coordinate retrieving the latest best model from the model registry, packaging the web application, deploying the web application, and inferencing web service. After a first successful deployment, you will make updates to both the model, the and web application, and execute the pipeline once to achieve an updated deployment. You will also learn how to monitor the model's performance after it is deployed, so WWI can be proactive with performance issues.
```
Comment (sureshpaulraj) [inline Hands-on lab/HOL step-by-step - MLOps.md:97]: After a first successful deployment, you will make updates to both the model, the and web application, and execute the pipeline once to achieve an updated deployment.
```suggestion
In this hands-on lab, you will learn how Wide World Importers (WWI) can leverage Deep Learning technologies to scan through their vehicle specification documents to find compliance issues with new regulations. You will standardize the model format to ONNX and observe how this simplifies inference runtime code, enabling pluggability of different models and targeting a broad range of runtime environments, and most importantly, improves inferencing speed over the native model. You will build a DevOps pipeline to coordinate retrieving the latest best model from the model registry, packaging the web application, deploying the web application, and inferencing web service. After a first successful deployment, you will make updates to both the model, **the** and web application, and execute the pipeline once to achieve an updated deployment. You will also learn how to monitor the model's performance after it is deployed, so WWI can be proactive with performance issues.
```

Comment (sureshpaulraj) [reply Hands-on lab/HOL step-by-step - MLOps.md:97]: Just a minor typo in **the** bold font

File: Hands-on lab/HOL step-by-step - MLOps.md [hunk 2]
```diff
@@ -251,22 +266,10 @@ Duration: 20 minutes
 
     ![In Azure DevOps, Repos is selected from the left menu. In the mlops-quickstart screen the Import button is selected in the Import a repository section.](media/devops-project-03.png 'Azure DevOps Repos')
 
-2. Provide the following GitHub URL: `https://github.com/solliancenet/mcw-mlops-starter-v2` and select **Import**. This should import the code required for the quickstart.
+2. Provide the following GitHub URL: `https://github.com/solliancenet/mcw-mlops-starter-v3` and select **Import**. This should import the code required for the quickstart.
```
Comment (sureshpaulraj) [inline Hands-on lab/HOL step-by-step - MLOps.md:269]: Don't know if the mcw-mlops-starter-v3/scripts/evaluate.py  is included for review. Just a minor cleanup.
The code "from azureml.pipeline.core import PipelineRun" is not needed (previous use of the module is commented)

**Comment by nchandhi** [comment]
Hi Jitendra, I remember Nishant showed this example during review to have QA deployment to ACI and Prod deployment to AKS (We dont have Synapse in this example so that's not needed)
image

But when I run this workshop, I only get one stage. Am I missing something? 
image

Thanks,
Nalini.

**Comment by nchandhi** [comment]
I saw a note in September proposed changes about - "Accountability is a key aspect of the Responbile ML development. Demonstrate the documentation of the machine learning lifecycle with datasheets."

Did we consider this change? If yes, can you please point me to which section of the lab to review for this change?

Thanks,
Nalini

File: Hands-on lab/HOL step-by-step - MLOps.md (hunk 1 above)
Comment (shirolkar) [reply Hands-on lab/HOL step-by-step - MLOps.md:97]: @sureshpaulraj I have made the typo fix for the second block where you highlighted the word **the**. I am not sure what typo is in the first block where you highlight Wide World Importers (WWI)?

File: Hands-on lab/HOL step-by-step - MLOps.md (hunk 2 above)
Comment (shirolkar) [reply Hands-on lab/HOL step-by-step - MLOps.md:269]: There was a change in Azure ML SDK in the middle of the update and it changed a behavior. I am thinking it’s a bug in the SDK. I had to make an emergency fix, and the commented lines are the original code. I would like to revert back to the original code when the Azure ML SDK issue is resolved in the near future.

**Comment by shirolkar** [comment]
> Hi Jitendra, I remember Nishant showed this example during review to have QA deployment to ACI and Prod deployment to AKS (We dont have Synapse in this example so that's not needed)
> image
> 
> But when I run this workshop, I only get one stage. Am I missing something?
> image
> 
> Thanks,
> Nalini.

image


> Hi @shirolkar , I remember Nishant showed this example during review to have QA deployment to ACI and Prod deployment to AKS (We dont have Synapse in this example so that's not needed)
> image
> 
> But when I run this workshop, I only get one stage. Am I missing something?
> image
> 
> Thanks,
> Nalini.

@nchandhi 
Hi Nalini – We setup two release pipelines (Exercise 5 and Exercise 6) – test (ACI) and production (AKS). The test pipeline is continuous deployment triggered based on new model registration and the production pipeline is continuous deployment triggered with manual approval based on successful build completion. It is a slightly different arrangement than what you show. Hope this clarifies. Thanks. Jitendra

**Comment by shirolkar** [comment]
> I saw a note in September proposed changes about - "Accountability is a key aspect of the Responbile ML development. Demonstrate the documentation of the machine learning lifecycle with datasheets."
> 
> Did we consider this change? If yes, can you please point me to which section of the lab to review for this change?
> 
> Thanks,
> Nalini

@nchandhi Hi Nalini - Please refer to Exercise 8: Testing the deployed solution and review deployed model datasheet section and the associated notebook. Thanks. Jitendra

**Pull request merged**